src/metabase/driver/sql/query_processor.clj

```diff
@@ -353,14 +353,15 @@
 
 (defmethod ->honeysql [:sql :field]
   [driver [_ id-or-name {:keys             [database-type]
-                         ::nest-query/keys [outer-select]
                          :as               options}
            :as field-clause]]
   (try
     (let [source-table-aliases (field-source-table-aliases field-clause)
           source-alias         (field-source-alias field-clause)
           field                (when (integer? id-or-name)
                                  (qp.store/field id-or-name))
+          outer-select         (or (:metabase.query-processor.middleware.mark-outer-select-fields/outer-select options)
```

Comment:
I don't like this name. Since it's not SQL specific, `select` doesn't sound well. (I will `:as-alias` the namespace, I think, when I get the name right.)

Reply:
Should we unify the keywords? Might as well replace all usages of `::nest-query/outer-select` with the new `mark-outer-select-fields/outer-select`? Or is that annoying for now or planned follow up work?

Reply:
If we agree that the meaning is the same, then we should use the same key, I think.

Reply:
Yes I believe these mean the same thing: a conversion has been applied already so don't convert again. We can also use a non-namespace aliased key `:nested/outer` or `:nested/inner` so we don't need to worry about requiring namespaces for the namespace of a key. 

If I'm misreading this then I apologize.